### PR TITLE
Abstract SignerConfigurator + PopUpConfigurator

### PR DESCRIPTION
**Pull request by fan-zhang-sv**
### _Summary_

Added 2 abstract layers

### _How did you test your changes?_

Manually

### CODE REVIEW

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 1]
```diff
@@ -36,6 +35,10 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
     return this.chain.id;
   }
 
+  public get isCoinbaseWallet() {
+    return true;
+  }
```
Comment (nateReiners) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:40]: thanks for this

**Review by nateReiners — APPROVED** [review]
Signer Configurator looks good
[non blocking nit] I'm not sure about the PopupCommunicator / PopUpConfigurator distinction/naming but we can iterate later if needed 🚢

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts (hunk 1 above)
Comment (nathweetwel-star) [reply packages/wallet-sdk/src/CoinbaseWalletProvider.ts:40]: > thanks for this

Comment (nathweetwel-star) [reply packages/wallet-sdk/src/CoinbaseWalletProvider.ts:40]: https://github.com/coinbase/coinbase-wallet-sdk/pull/1136#pullrequestreview-1950146651

Comment (nathweetwel-star) [reply packages/wallet-sdk/src/CoinbaseWalletProvider.ts:40]: https://github.com/coinbase/coinbase-wallet-sdk/pull/1136#pullrequestreview-1950146651

**Pull request merged**